client/my-sites/customer-home/controller.jsx

```diff
@@ -29,21 +49,20 @@ export function maybeRedirect( context, next ) {
 	}
 
 	const siteId = getSelectedSiteId( state );
-	const options = getSiteOptions( state, siteId );
 
-	// Normally, checking the launchpad_screen option in redux state would be enough to decide whether
-	// or not to redirect to launchpad. The option, however, is loading stale data in horizon, and presumably,
-	// in production as well. The forceLoadLaunchpadData query param is a temporary patch to circumvent stale data, and
-	// will avoid a redirect.
-	const shouldRedirectToLaunchpad =
-		options?.launchpad_screen === 'full' &&
-		! getQueryArg( window.location.href, 'forceLoadLaunchpadData' );
+	if ( getSiteOptions( state, siteId )?.launchpad_screen !== 'off' ) {
+		await refetchLaunchpadScreenOption( siteId, context );
+	}
```

Comment:
It looks like the idea here is to limit awaiting a refetch if the sites already have the launchpad turned off. Two thoughts:

1. We probably want to check something like `if ( launchpad_screen && launchpad_screen !== 'off' )` to cover the falsy values of there being no launchpad screen setting altogether I think? Otherwise we will be forcing every site created before the launchpad and not affiliated with it to await this refetch.
2. (non-blocking) This type of logic makes sense for now, but thinking for the possible future could pose problems the other direction - if we get to the point where we are enabling the launchpad on existing sites selecting a new goal then this form of limiting the request seems like it would cause us to have the opposite issue (where my home doesn't redirect to launchpad when it should due to the stale data). Although that is much further down the horizon and less certain of a flow (maybe that stays encapsulated in my home anyways) - so maybe not a concern but good to be aware of in the future.

Reply:
> We probably want to check something like if ( launchpad_screen && launchpad_screen !== 'off' ) to cover the falsy values of there being no launchpad screen setting altogether I think? Otherwise we will be forcing every site created before the launchpad and not affiliated with it to await this refetch.

Ah yes. You're totally right. That'll be a quick fix we push in a bit. Thanks for pointing that out.

> if we get to the point where we are enabling the launchpad on existing sites selecting a new goal then this form of limiting the request seems like it would cause us to have the opposite issue (where my home doesn't redirect to launchpad when it should due to the stale data).

Good point. I hadn't considered this scenario 🤔

I agree, and the problem would probably manifest as stale data for the first load of myHome ( at which point the `QueryAll` component rendered in React would query site information and invalidate the cache ). Not sure how we'd approach this problem -- totally get that it's non-blocking, but will still mull this over.

Reply:
[093a29e](https://github.com/Automattic/wp-calypso/pull/67967/commits/093a29e7a76e66f53e6bc29a93d65e471087e43e)